jetty-core/jetty-compression/jetty-compression-server/src/main/java/org/eclipse/jetty/compression/server/CompressionHandler.java

```diff
@@ -304,6 +303,26 @@ public String toString()
     @Override
     protected void doStart() throws Exception
     {
+        // If the supported encodings is empty, that means this handler wasn't manually configured with encodings.
+        // Fallback to discovered encodings via the service loader instead.
+        if (supportedEncodings.isEmpty())
+        {
+            TypeUtil.serviceProviderStream(ServiceLoader.load(Compression.class)).forEach(
+                compressionProvider ->
+                {
+                    try
+                    {
+                        Compression compression = compressionProvider.get();
+                        addCompression(compression);
+                    }
+                    catch (Throwable e)
+                    {
+                        LOG.warn("Unable to get Compression", e);
```

Comment:
Use `debug` level here, but better yet, use `serviceStream()` and get rid of try/catch.